compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -186,6 +223,20 @@ struct LinalgFusePass : public LinalgFuseBase<LinalgFusePass> {
   void runOnOperation() override;
 };
 
+struct LinalgSplitReductionPass
+    : public LinalgSplitReductionBase<LinalgSplitReductionPass> {
+  LinalgSplitReductionPass(bool enableReductionReordering, int64_t size = 0,
+                           uint64_t unrollFactor = 1) {
+    this->size.setValue(size);
+    this->unrollFactor.setValue(unrollFactor);
+    reductionReordering = enableReductionReordering;
+  }
+  void runOnOperation() override;
+
+ private:
+  bool reductionReordering = true;
```

Comment:
We can't because reductionReordering is *always* enabled if the operands of the pass is integers. It's not enabled *only if* the operands are floats or other data types where reductionReordering can lead to different results

Reply:
Can we resolve this comment?

Reply:
I think having both is fine. Unfortunately we need both pipeline level flag and pass level flag (latter at least for testing). It makes sense to use the pipeline level flag to set the flag of the pass. This is what MLIR is moving towards as well.

Ill resolve this comment.

Reply:
Thanks for the explanation, I now understand what we want to achieve. Can we rename it to reductionFPReordering? The integer cases are always reordered in the pass, and the FP cases are reordered only when the flag is true?

Reply:
> Can we rename it to reductionFPReordering?
It's called clEnableReassociateFpReductions based on what @dcaballe suggested.

> The integer cases are always reordered in the pass, and the FP cases are reordered only when the flag is true?
Yes